src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/commands/mobile/subscribe_vehicle_data_request.cc

```diff
@@ -201,20 +201,39 @@ bool SubscribeVehicleDataRequest::CheckSubscriptionStatus(
 }
 
 bool SubscribeVehicleDataRequest::SubscribePendingVehicleData(
-    ApplicationSharedPtr app, const smart_objects::SmartObject& msg_params) {
+    ApplicationSharedPtr app, smart_objects::SmartObject& msg_params) {
   LOG4CXX_DEBUG(logger_, "Subscribing to all pending VehicleData");
 
-  for (const auto& vi_name : vi_waiting_for_subscribe_) {
+  std::set<hmi_apis::Common_VehicleDataResultCode::eType> skiped_result_codes(
+      {hmi_apis::Common_VehicleDataResultCode::VDRC_TRUNCATED_DATA,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_DISALLOWED,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_USER_DISALLOWED,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_INVALID_ID,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_DATA_NOT_AVAILABLE,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_DATA_NOT_SUBSCRIBED,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_IGNORED});
+
+  for (auto vi_name = vi_waiting_for_subscribe_.begin();
```

Comment:
Can we replace this cycle with foreach like:
`for(auto& vi_name : vi_waiting_for_subscribe_) `?

Reply:
I think, no, because it may cause strange behavior, and if we will use range-based for, sometimes it may work good, and sometimes not. Instead, proposed way of iterating is safe, because in this case we don't need to bother about correct order of iterators after possible call of erase(). This approach is common for cases with possible erasing from associative containers like set.